### PR TITLE
added max function for sheet to determine for example the highest mou…

### PR DESCRIPTION
**Pull request by michaone007**
…ntain

### CODE REVIEW

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 1]
```diff
@@ -74,7 +74,39 @@ public static String chooseStringFromField(HashMap<String, Object> hash, String
...
+        int max = 0;
+        for (String s: answers){
+            if (Integer.valueOf(s) > max) {
+                max = Integer.valueOf(s);
+            }
+        }
+        int n = getRightMaxAnswer(answers,max);
+        String rightAnswer = answers2.get(n);
+        Log.d(TAG, "max = " + max);
+        return new TriviaQuestion("someKey", "מהו ההר הכי גבוה?",rightAnswer, answers2.get(0), answers2.get(1), answers2.get(2), answers2.get(3));
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: the question should be an argument of this method, not hard coded here.

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 2]
```diff
@@ -74,7 +74,39 @@ public static String chooseStringFromField(HashMap<String, Object> hash, String
         Collections.shuffle(lst);
         return lst.get(0);
     }
-
+    public static TriviaQuestion maxBuildQuestionHashFromData(ArrayList<HashMap<String, Object>> data, String answerKey,String questionKey) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: lets make this method for max && min, just add another argument to indicate.

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 3]
```diff
@@ -74,7 +74,39 @@ public static String chooseStringFromField(HashMap<String, Object> hash, String
...
+            if (Integer.valueOf(s) > max) {
+                max = Integer.valueOf(s);
+            }
+        }
+        int n = getRightMaxAnswer(answers,max);
+        String rightAnswer = answers2.get(n);
+        Log.d(TAG, "max = " + max);
+        return new TriviaQuestion("someKey", "מהו ההר הכי גבוה?",rightAnswer, answers2.get(0), answers2.get(1), answers2.get(2), answers2.get(3));
+    }
+    public static int getRightMaxAnswer(ArrayList<String> ans,int m) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java:126]: I'm not sure we need this. can't we just find the max/min above without it?

File: app/src/main/java/com/example/zakiva/santa/Models/Generator.java [hunk 4]
```diff
@@ -79,6 +79,9 @@ public TriviaQuestion countryToCapital () {
     public TriviaQuestion inventionToInventor() {
         return GeneratorHelper.generateQuestionWithData(TriviaGame.dataHash.get("inventions"), "מי המציא את #$#?", "name", "inventor");
     }
+    public TriviaQuestion highestMountain() {
+        return GeneratorHelper.maxBuildQuestionHashFromData(TriviaGame.dataHash.get("mountains"),"name","height (m)");
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/Generator.java]: as mentioned above, pass from here the question string.

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java (hunk 3 above)
Comment (michaone007) [reply app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java:126]: well i have the max number that is a integer but i need to know its name i can try to think about a different name

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java (hunk 1 above)
Comment (michaone007) [reply app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: yeah to make more generic

**Comment by zakiva** [comment]
unique answers, hashes with sort in array

**Pull request closed without merging**